types/webpack/index.d.ts

```diff
@@ -142,6 +136,16 @@ declare namespace webpack {
         optimization?: Options.Optimization;
     }
 
+    type ConfigurationFactory = ((
+        env: string | Record<string, boolean | number | string>,
+        args: Record<string, string>,
```

Comment:
@rubenspgcavalcante, what about extending `args` with known values? I don't know if there are any other ones, but the following would fit my needs:

```ts
args: Record<string, string> & { mode?: "development" | "production" | "none" },
```

(Obviously exported as its type to be reused for `MultiConfigurationFactory` as well)

We can see at https://webpack.js.org/configuration/mode/ that `args.mode` is expected to be there.

Reply:
FYI, I opened https://github.com/DefinitelyTyped/DefinitelyTyped/pull/40262 with this. 🤞